src/Raven.Server/ServerWide/Maintenance/Sharding/OrchestratorTopologyUpdater.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using Raven.Client.ServerWide;
+using Raven.Server.Config.Categories;
+using Raven.Server.Rachis;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Utils;
+
+namespace Raven.Server.ServerWide.Maintenance.Sharding
+{
+    internal class OrchestratorTopologyUpdater : DatabaseTopologyUpdater
+    {
+        public OrchestratorTopologyUpdater(ServerStore server,
+            RachisConsensus<ClusterStateMachine> engine,
+            ClusterConfiguration clusterConfiguration,
+            DateTime clusterObserverStartTime,
+            ObserverLogger logger) : base(server, engine, clusterConfiguration, clusterObserverStartTime, logger)
+        {
+        }
+
+        protected override (bool Promote, string UpdateTopologyReason) TryGetMentorAndPromote(ClusterOperationContext context, ClusterObserver.DatabaseObservationState state, string promotable)
+        {
+            return (true, $"Node {promotable} is ready to be promoted to orchestrator");
+        }
+
+        protected override bool IsLastCommittedIndexCaughtUp(ClusterOperationContext context, string node, DatabaseTopology topology, ClusterNodeStatusReport nodeStats, long iteration)
```

Comment:
shouldn't this be part of `TryGetMentorAndPromote` ?

Reply:
Changed